File: Newbie_basics/Python/addDigits.py
class Solution:
    def addDigits(self, num: int) -> int:
        while num > 9:
            num = sum([int(i) for i in str(num)])
        return num

def main():
    solution = Solution()
    #user_input = int(input())
    result = solution.addDigits(38)
    result2 = solution.addDigits(0)
    print(result)
    print(result2)

if __name__ == "__main__":
    main()

File: Newbie_basics/C++/addDigits.cpp
#include <iostream>
#include <string>
using namespace std;
class Solution {
    public:
        int addDigits(int num) {
            while (num >= 10){
            int sum = 0;
            while (num >0) {
                sum += num % 10;
                num /= 10;
            }
            num = sum; 
            }
            return num;
        }
    };

int main() {
    Solution solution;
    cout << solution.addDigits(38) << endl;
    cout << solution.addDigits(0) << endl;
    return 0;
}
        
